pxr/usdImaging/usdImagingGL/engine.h

```diff
@@ -316,6 +316,33 @@ class UsdImagingGLEngine
         int *outHitInstanceIndex = NULL,
         HdInstancerContext *outInstancerContext = NULL);
 
+    struct IntersectionResult
+    {
+        SdfPath hitPrimPath;
+        SdfPath hitInstancerPath;
+        int hitInstanceIndex = 0;
+        HdInstancerContext* instancerContext = NULL;
+    };
+
+    typedef std::vector<struct IntersectionResult> IntersectionResultVector;
+
+    /// Finds all objects intersecting a frustum by rendering.
+    ///	
+    /// This method uses Deep Selection to gather all paths within a region even 
+    /// if obscured by other visible objects.
+    ///
+    /// Returns whether a hit occurred and if so, \p outResults will point to all the 
+    /// gprims selected by the pick. \p outHitInstancerPath will point to the point 
+    /// instancer (if applicable) of each gprim. 
+    ///
+    USDIMAGINGGL_API
+    bool TestIntersections(
```

Comment:
We've been talking about a new TestIntersection API, unrelated to this change.  If you want to do us a favor, if you call this "TestIntersection" and take a "const TfToken& resolveMode", and swap out the "const UsdPrim& root" for a "const SdfPathVector&" (like RenderBatch), we can call that the new intersection API, and I can mark the existing one as deprecated.

Reply:
Also: if you can bump the version in usdImagingGL/version.h, that would be great.